File: client-scalaz/src/main/scala/org/openapitools/client/api/ConnectionInfoResponse.scala
package org.openapitools.client.api

import argonaut._
import argonaut.EncodeJson._
import argonaut.DecodeJson._

import org.http4s.{EntityDecoder, EntityEncoder}
import org.http4s.argonaut._
import org.joda.time.DateTime


import ConnectionInfoResponse._

case class ConnectionInfoResponse (
  valid: Option[Boolean],
message: Option[String],
version: Option[Double],
serializeNulls: Option[Boolean],
startTimeLog: Option[Long],
errorCode: Option[String],
request: Option[List[NameStringValueResponse]],
groups: Option[List[ConnectionGroupResponse]],
followerCount: Option[Integer],
followingCount: Option[Integer],
friendCount: Option[Integer],
privateGroupCount: Option[Integer],
returning: Option[String])

object ConnectionInfoResponse {
  import DateTimeCodecs._

  implicit val ConnectionInfoResponseCodecJson: CodecJson[ConnectionInfoResponse] = CodecJson.derive[ConnectionInfoResponse]
  implicit val ConnectionInfoResponseDecoder: EntityDecoder[ConnectionInfoResponse] = jsonOf[ConnectionInfoResponse]
  implicit val ConnectionInfoResponseEncoder: EntityEncoder[ConnectionInfoResponse] = jsonEncoderOf[ConnectionInfoResponse]
}

File: client-clojure/src/sirqul_io_t_platform/specs/connection_info_response.clj
(ns sirqul-io-t-platform.specs.connection-info-response
  (:require [clojure.spec.alpha :as s]
            [spec-tools.data-spec :as ds]
            [sirqul-io-t-platform.specs.name-string-value-response :refer :all]
            [sirqul-io-t-platform.specs.connection-group-response :refer :all]
            )
  (:import (java.io File)))


(def connection-info-response-data
  {
   (ds/opt :valid) boolean?
   (ds/opt :message) string?
   (ds/opt :version) float?
   (ds/opt :serializeNulls) boolean?
   (ds/opt :startTimeLog) int?
   (ds/opt :errorCode) string?
   (ds/opt :request) (s/coll-of name-string-value-response-spec)
   (ds/opt :groups) (s/coll-of connection-group-response-spec)
   (ds/opt :followerCount) int?
   (ds/opt :followingCount) int?
   (ds/opt :friendCount) int?
   (ds/opt :privateGroupCount) int?
   (ds/opt :returning) string?
   })

(def connection-info-response-spec
  (ds/spec
    {:name ::connection-info-response
     :spec connection-info-response-data}))
